### ISSUE TITLE
Consider writing a setup script and get py2exe working

### ISSUE THREAD

**Issue by cyberkitsune** [cyberkitsune/vrc-osc-scripts#3]
Compiling out an actual exe for both VRCSubs and VRCNowPlaying may be nice, making it so people won't have to install python / click the add to PATH checkbox.

**Comment by senkodev** [comment]
Why not PyInstaller?

**Comment by MaxFerney** [comment]
I've used cx_Freeze to make python executables about 5 years ago, Not sure if it's still good for it, but it makes pretty neatly packaged executables and is not too hard to setup.

**Comment by KuryKat** [comment]
To be honest it is a good idea but at the same time people have been very against using """random""" .exe (after all Discord scams)

**Comment by cyberkitsune** [comment]
> To be honest it is a good idea but at the same time people have been very against using """random""" .exe (after all Discord scams)

Yeah this has been one of the main blockers to this, doubly so when a lot of malicious software does get wrapped with stuff like py2exe and trips up antivirus stuff all the time.

I'm still considering it though, as python installation issues are 90% of the support requests I get. I may not move to this until I can sign exes though...

**Comment by ProfessionalNincompoop** [comment]
Here's another potential solution, i thought I'd weigh in with, bundle the repository with the start.bat's and have those instead of launching python from path, launch from a virtual env skipping all the downloading and also preventing things from breaking when a package updates, I'm on my phone and can't test if this would work but i see no reason why it wouldn't because virtual envs or at least everyone I've created all have their own bundled python binaries, and any Linux users almost certainly know how to install a requirements.py if we just leave it there

**Comment by ProfessionalNincompoop** [comment]
So after much pain and research, ive discovered 2 things I hate git and it does everything in its power to make uploading a venv hell and venvs do not work across computers.
Yes I made sure it wasn't in git ignore

**Comment by cyberkitsune** [comment]
Yeah I don't think shipping a venv would be wise, they're not quite designed for distribution. I think some python to executable wrapper will be the best way to go about this. I'll mess around and evaluate a few of them and see what works best ^^
